wrappers/adaptive/popcorn.HTMLAdaptiveMediaElement.js

```diff
@@ -85,197 +102,213 @@
-                        q.resolution = q.height;
-                        q.value = idx;
-                        return q;
-                      });
-                      bitrates.push({ resolution: autoQualityName, value: autoQuality });
-                      media.qualities = bitrates;
-                    } else {
-                      media.qualities = [];
-                    }
-                    media.currentQuality = player.getQualityFor('video');
-                    media.dispatchEvent("bitrateloaded");
-                    updateQuality = function(currentQuality) {
-                      if (currentQuality === autoQuality) {
-                        player.setAutoSwitchQualityFor('video', true);
-                      } else {
-                        player.setAutoSwitchQualityFor('video', false);
-                        player.setQualityFor('video', currentQuality);
-                      }
-                    }
-                  });
-                  player.initialize(media, adaptiveMedia, false);
-                });
-                break;
-              case 'm3u8':
-                loadHlsJs(media, function(hls) {
-                  if(Hls.isSupported()) {
-                    hls.on(Hls.Events.ERROR, function(error, data) {
-                      // fallback to default media source
-                      if (data.type === 'networkError') {
+            if (dashMedia || hlsMedia) {
+              var adaptiveMedia = dashMedia || hlsMedia;
+              var extension = adaptiveMedia.split('.').reverse()[0];
+              switch (extension) {
+                case 'mpd':
+                  loadDashJs(function() {
+                    var player = dashjs.MediaPlayer().create();
+                    player.on(dashjs.MediaPlayer.events.ERROR, function(event) {
+                      if (event.error === 'capability') {
+                        // 23 says `message: "mediasource is not supported"`, so fallback to HLS
+                        // as it happens mainly on Safari iOS
+                        media.src = hlsMedia || fallbackMedia;
+                      } else if (event.error === 'download' && event.event.id === 'manifest') {
+                        // otherwise MPD manifest is not available so fallback to regular media file
                         media.src = fallbackMedia;
                       }
                     });
-                    hls.on(Hls.Events.MEDIA_ATTACHED, function() {
-                      var bitrates = hls.levels;
+                    player.on(dashjs.MediaPlayer.events.SOURCE_INITIALIZED, function() {
+                      player.setTrackSwitchModeFor('video', 'alwaysReplace');
+                      player.setTrackSwitchModeFor('audio', 'alwaysReplace');
+                      player.setAutoSwitchQualityFor('video', true);
+                      player.setAutoSwitchQualityFor('audio', true);
+                      player.setInitialBitrateFor('audio', 99999999);
```

Comment:
Why `99999999 ` ?

Reply:
@TheRadly It's old code, I simple move it.